// mondaine-clock-react/dist/cjs/demo.js
"use strict";
exports.__esModule = true;
exports.Demo = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importStar(require("react"));
var mondaine_clock_1 = require("./components/mondaine-clock");
var date = new Date();
var Demo = function () {
    var defaultRed = '#eb0000';
    var defaultHand = '#0e0e10';
    var defaultBackground = '#ffffff';
    var _a = (0, react_1.useState)(mondaine_clock_1.TickType.MECHANICAL), tickType = _a[0], setTickType = _a[1];
    var _b = (0, react_1.useState)(false), sizeAuto = _b[0], setSizeAuto = _b[1];
    var _c = (0, react_1.useState)(200), size = _c[0], setSize = _c[1];
    var _d = (0, react_1.useState)(false), fixDateEnabled = _d[0], setFixDateEnabled = _d[1];
    var _e = (0, react_1.useState)(date), fixedDate = _e[0], setFixedDate = _e[1];
    var _f = (0, react_1.useState)(true), showSecondsHand = _f[0], setShowSecondsHand = _f[1];
    var _g = (0, react_1.useState)(true), showCenterDot = _g[0], setShowCenterDot = _g[1];
    var _h = (0, react_1.useState)(false), showLogo = _h[0], setShowLogo = _h[1];
    var _j = (0, react_1.useState)(false), customColors = _j[0], setCustomColors = _j[1];
    var _k = (0, react_1.useState)(defaultRed), secondsHandColor = _k[0], setSecondsHandColor = _k[1];
    var _l = (0, react_1.useState)(defaultHand), minutesHandColor = _l[0], setMinutesHandColor = _l[1];
    var _m = (0, react_1.useState)(defaultHand), hoursHandColor = _m[0], setHoursHandColor = _m[1];
    var _o = (0, react_1.useState)(defaultHand), faceColor = _o[0], setFaceColor = _o[1];
    var _p = (0, react_1.useState)(defaultBackground), faceBackgroundColor = _p[0], setFaceBackgroundColor = _p[1];
    var _q = (0, react_1.useState)(false), copySuccess = _q[0], setCopySuccess = _q[1];
    (0, react_1.useEffect)(function () {
        var timeout = null;
        if (copySuccess) {
            timeout = setTimeout(function () {
                setCopySuccess(false);
            }, 2000);
        }
        return function () {
            if (timeout) {
                clearTimeout(timeout);
            }
        };
    }, [copySuccess]);
    function handleReset() {
        setTickType(mondaine_clock_1.TickType.MECHANICAL);
        setSizeAuto(false);
        setSize(200);
        setFixDateEnabled(false);
        setFixedDate(new Date());
        setCustomColors(false);
        setShowSecondsHand(true);
        setShowCenterDot(true);
        setShowLogo(false);
        setSecondsHandColor(defaultRed);
        setMinutesHandColor(defaultHand);
        setHoursHandColor(defaultHand);
        setFaceColor(defaultHand);
        setFaceBackgroundColor(defaultBackground);
    }
    function handleResetColors() {
        setSecondsHandColor(defaultRed);
        setMinutesHandColor(defaultHand);
        setHoursHandColor(defaultHand);
        setFaceColor(defaultHand);
        setFaceBackgroundColor(defaultBackground);
    }
    var formatTime = function (date) {
        var hours = date.getHours().toString().padStart(2, '0');
        var minutes = date.getMinutes().toString().padStart(2, '0');
        var seconds = date.getSeconds().toString().padStart(2, '0');
        return "".concat(hours, ":").concat(minutes, ":").concat(seconds);
    };
    // Event handler for input change
    var handleTimeChange = function (e) {
        var inputTime = e.target.value;
        var _a = inputTime.split(':').map(Number), hours = _a[0], minutes = _a[1], seconds = _a[2];
        if (!isNaN(hours) && !isNaN(minutes) && !isNaN(seconds)) {
            var newDate = new Date();
            newDate.setHours(hours);
            newDate.setMinutes(minutes);
            newDate.setSeconds(seconds);
            newDate.setMilliseconds(0);
            setFixedDate(newDate);
        }
    };
    return (react_1["default"].createElement("div", { style: {
            fontFamily: 'Helvetica, Arial, sans-serif'
        } },
        react_1["default"].createElement("table", { style: {
                textAlign: 'left',
                background: "whitesmoke",
                width: '100%'
            } },
            react_1["default"].createElement("tbody", null,
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", { style: { width: '200px' } }, "Clock Type"),
                    react_1["default"].createElement("td", { style: { width: '200px' } },
                        react_1["default"].createElement("select", { style: { width: '200px' }, value: tickType, onChange: function (e) { return setTickType(parseInt(e.target.value)); } },
                            react_1["default"].createElement("option", { value: mondaine_clock_1.TickType.MECHANICAL }, "Mechanical"),
                            react_1["default"].createElement("option", { value: mondaine_clock_1.TickType.QUARTZ }, "Quartz"))),
                    react_1["default"].createElement("td", null, tickType === mondaine_clock_1.TickType.MECHANICAL &&
                        fixDateEnabled &&
                        'Using Quartz Clock is recommended when using a fixed date.')),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Size Auto"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { type: "checkbox", checked: sizeAuto, onChange: function (e) { return setSizeAuto(e.target.checked); } }))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Size"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { style: { width: '200px' }, type: "range", min: 100, max: 800, value: size, onChange: function (e) { return setSize(parseInt(e.target.value)); }, disabled: sizeAuto })),
                    react_1["default"].createElement("td", null,
                        size,
                        "px")),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Fixed Date enabled"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { type: "checkbox", checked: fixDateEnabled, onChange: function (e) { return setFixDateEnabled(e.target.checked); } }))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Date"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { style: { width: '200px' }, type: "time", step: "1", value: formatTime(fixedDate), onChange: handleTimeChange, disabled: !fixDateEnabled }))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Show Seconds Hand"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { type: "checkbox", checked: showSecondsHand, onChange: function (e) { return setShowSecondsHand(e.target.checked); } }))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Show Center Dot"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { type: "checkbox", checked: showCenterDot, onChange: function (e) { return setShowCenterDot(e.target.checked); } }))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Show Logo"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { type: "checkbox", checked: showLogo, onChange: function (e) { return setShowLogo(e.target.checked); } }))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null, "Custom Colors"),
                    react_1["default"].createElement("td", null,
                        react_1["default"].createElement("input", { type: "checkbox", checked: customColors, onChange: function (e) {
                                var checked = e.target.checked;
                                if (!checked) {
                                    handleResetColors();
                                }
                                setCustomColors(checked);
                            } }))),
                customColors && (react_1["default"].createElement(react_1["default"].Fragment, null,
                    react_1["default"].createElement("tr", null,
                        react_1["default"].createElement("th", null, "Seconds Hand Color"),
                        react_1["default"].createElement("td", null,
                            react_1["default"].createElement("input", { style: { width: '200px' }, type: "color", value: secondsHandColor, onChange: function (e) { return setSecondsHandColor(e.target.value); } })),
                        react_1["default"].createElement("td", null, secondsHandColor)),
                    react_1["default"].createElement("tr", null,
                        react_1["default"].createElement("th", null, "Minutes Hand Color"),
                        react_1["default"].createElement("td", null,
                            react_1["default"].createElement("input", { style: { width: '200px' }, type: "color", value: minutesHandColor, onChange: function (e) { return setMinutesHandColor(e.target.value); } })),
                        react_1["default"].createElement("td", null, minutesHandColor)),
                    react_1["default"].createElement("tr", null,
                        react_1["default"].createElement("th", null, "Hours Hand Color"),
                        react_1["default"].createElement("td", null,
                            react_1["default"].createElement("input", { style: { width: '200px' }, type: "color", value: hoursHandColor, onChange: function (e) { return setHoursHandColor(e.target.value); } })),
                        react_1["default"].createElement("td", null, hoursHandColor)),
                    react_1["default"].createElement("tr", null,
                        react_1["default"].createElement("th", null, "Face Color"),
                        react_1["default"].createElement("td", null,
                            react_1["default"].createElement("input", { style: { width: '200px' }, type: "color", value: faceColor, onChange: function (e) { return setFaceColor(e.target.value); } })),
                        react_1["default"].createElement("td", null, faceColor)),
                    react_1["default"].createElement("tr", null,
                        react_1["default"].createElement("th", null, "Face Background Color"),
                        react_1["default"].createElement("td", null,
                            react_1["default"].createElement("input", { style: { width: '200px' }, type: "color", value: faceBackgroundColor, onChange: function (e) { return setFaceBackgroundColor(e.target.value); } })),
                        react_1["default"].createElement("td", null, faceBackgroundColor)))),
                react_1["default"].createElement("tr", null,
                    react_1["default"].createElement("th", null,
                        react_1["default"].createElement("button", { onClick: handleReset }, "Reset"))))),
        react_1["default"].createElement("div", null,
            react_1["default"].createElement(mondaine_clock_1.MondaineClock, { tickType: tickType, width: sizeAuto ? undefined : "".concat(size, "px"), fixedDate: fixDateEnabled ? fixedDate : undefined, showSecondsHand: showSecondsHand, showCenterDot: showCenterDot, showLogo: showLogo, secondsHandColor: secondsHandColor, minutesHandColor: minutesHandColor, hoursHandColor: hoursHandColor, faceColor: faceColor, faceBackgroundColor: faceBackgroundColor })),
        react_1["default"].createElement("pre", { id: "code", style: {
                background: "whitesmoke",
                fontFamily: 'consolas, monospace',
                fontSize: '12px',
                padding: '10px'
            }, onClick: function () {
                var _a, _b;
                var code = document.getElementById('code');
                if (code) {
                    var range = document.createRange();
                    range.selectNode(code);
                    (_a = window.getSelection()) === null || _a === void 0 ? void 0 : _a.removeAllRanges();
                    (_b = window.getSelection()) === null || _b === void 0 ? void 0 : _b.addRange(range);
                    setCopySuccess(true);
                }
            } },
            '<MondaineClock',
            react_1["default"].createElement("br", null), "    tickType={TickType.".concat(mondaine_clock_1.TickType[tickType], "}"),
            react_1["default"].createElement("br", null),
            sizeAuto ? null : "    width={'".concat(size, "px'}"),
            sizeAuto ? null : react_1["default"].createElement("br", null),
            fixDateEnabled ? "    fixedDate={new Date('".concat(fixedDate.toISOString(), "')}") : null,
            fixDateEnabled ? react_1["default"].createElement("br", null) : null,
            showSecondsHand ? null : "    showSecondsHand={false}",
            showSecondsHand ? null : react_1["default"].createElement("br", null),
            secondsHandColor === defaultRed ? null : "    secondsHandColor={'".concat(secondsHandColor, "'}"),
            secondsHandColor === defaultRed ? null : react_1["default"].createElement("br", null),
            showCenterDot ? null : "    showCenterDot={false}",
            showCenterDot ? null : react_1["default"].createElement("br", null),
            !showLogo ? null : "    showLogo={true}",
            !showLogo ? null : react_1["default"].createElement("br", null),
            minutesHandColor === defaultHand ? null : "    minutesHandColor={'".concat(minutesHandColor, "'}"),
            minutesHandColor === defaultHand ? null : react_1["default"].createElement("br", null),
            hoursHandColor === defaultHand ? null : "    hoursHandColor={'".concat(hoursHandColor, "'}"),
            hoursHandColor === defaultHand ? null : react_1["default"].createElement("br", null),
            faceColor === defaultHand ? null : "    faceColor={'".concat(faceColor, "'}"),
            faceColor === defaultHand ? null : react_1["default"].createElement("br", null),
            faceBackgroundColor === defaultBackground ? null : "    faceBackgroundColor={'".concat(faceBackgroundColor, "'}"),
            faceBackgroundColor === defaultBackground ? null : react_1["default"].createElement("br", null),
            '/>'),
        copySuccess ? react_1["default"].createElement("span", { style: { color: 'green' } }, "Copied!") : null,
        react_1["default"].createElement("p", null,
            react_1["default"].createElement("small", null,
                "By ",
                react_1["default"].createElement("a", { href: "https://github.com/orbitertoad", target: '_blank', rel: "noreferrer" }, "OrbiterToad")))));
};
exports.Demo = Demo;
